internal/tools/changes/repository_test.go

```diff
@@ -0,0 +1,102 @@
+package changes
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func TestNewRepository(t *testing.T) {
+	repo, err := NewRepository("testdata")
+	if err != nil {
+		t.Error(err)
+	}
+
+	if repo.Metadata.ChangePath != "testdata/.changes" {
+		t.Errorf("expected Metadata.ChangePath to be testdata/.changes, got %s", repo.Metadata.ChangePath)
+	}
+}
+
+func TestRepository_UpdateChangelog(t *testing.T) {
+	repo := getRepository(t)
+
+	for _, changelogType := range []string{"pending", "regular"} {
+		cases := getTestChangelogCases(t, changelogType)
+		for _, tt := range cases {
+			pending := false
+			fileName := "testdata/CHANGELOG.md"
+			if changelogType == "pending" {
+				pending = true
+				fileName = "testdata/CHANGELOG_PENDING.md"
+			}
```

Comment:
nit, these should use `filepath.Join` instead of inline `/`  These tests probably won't work on windows.